app/controllers/actions/AuthAction.scala

```diff
@@ -67,6 +73,22 @@ class AuthActionImpl @Inject()(val authConnector: AuthConnector, val config: Con
       case _                  => Left(Redirect(routes.UnauthorisedController.onPageLoad()))
     }
   }
+
+  private def isEoriWhitelisted(enrolments: Enrolments): Boolean = {
+    val eoriOpt = enrolments.getEnrolment("HMRC-CUS-ORG").flatMap(_.getIdentifier("EORINumber"))
+
+    eoriOpt.exists(eori => eoriWhitelist.allows(eori.value))
+  }
 }
 
 trait AuthAction extends ActionBuilder[AuthenticatedRequest, AnyContent] with ActionRefiner[Request, AuthenticatedRequest]
+
+@ProvidedBy(classOf[EoriWhitelistProvider])
```

Comment:
This google ProvidedBy thingy seems a bit over-kill.   You are injecting the config to get the list of white-listed Eoris.    Why not build your EoriWhitelist class using the config?

Reply:
Hmm, I copied this from Exports service

I can implement it in different way if you want :)

Reply:
Up to you.  Importing google.anything always makes me nervous for some reason.